test/scripts/roll_call_matrix.R

```diff
@@ -0,0 +1,119 @@
+# these variables, that **must** be named 'party', 'gov_coalition', and 
+# 'state'.
+# If Legis is TRUE, the object returned will be a list of two elements, 
+# the rollcall matrix/object/vector, and a dataframe of legislator 
+# characteristics. If you have a column of legislator names and wish to use 
+#this instead of IDs (for plotting purposes etc), simply name this column 
+# "Legis_ID". You will need the tidyverse package to run this.
+
+library(tidyverse)
+
+rollcallmatrix <- function(x, pscl = FALSE, Stan = FALSE, 
+                           Legis = FALSE, names = FALSE){
+  
+  if(pscl == TRUE & Stan == TRUE){
+    return(message("Error: arguments 'pscl' and 'Stan' cannot both be TRUE."))
+  }
+  
+  if(!require("pacman")) install.packages("pacman")
+  suppressMessages(library(pacman))
+  pacman::p_load(char=c("pscl"), install=TRUE)
+  
+  legislatorId <- x$Legis_ID
+  voteId <- x$Vote_ID
+  vote <- x$Vote
+  nameID <- unique(legislatorId)
+  n <- length(unique(nameID))
+  m <- length(unique(voteId))
+  
+  rollCallMatrix <- matrix(NA, n, m)
+  name_row <- match(legislatorId, nameID)
+  name_col <- unique(voteId)
+  name_col <- match(voteId, name_col)
+  
+  for(k in 1:length(legislatorId)){
+    rollCallMatrix[name_row[k], name_col[k]] <- vote[k]
+  }
+  
+  rollCallMatrix <- matrix(as.numeric(unlist(rollCallMatrix)), 
+                           nrow = nrow(rollCallMatrix))
+  dimnames(rollCallMatrix) <- list(unique(nameID), unique(voteId))
+  
+  
+  Leg <- data_frame(Legislator = unique(nameID),
+                    Party = x$party[match(unique(nameID),
+                                          x$Legis_ID)],
+                    State = x$state[match(unique(nameID),
+                                          x$Legis_ID)],
+                    GovCoalition = x$gov_coalition[match(unique(nameID),
+                                                         x$Legis_ID)])
+  
+  if(Legis == TRUE & pscl == FALSE & Stan == FALSE){
+    
+    Legis_data <- Leg
+    message("Returning rollcall matrix and legislator characteristics")
+    list_m <- list(rollCallMatrix = rollCallMatrix, 
+                   Legis_data = Legis_data)
+    return(list_m)
+    
+  } else if(Legis == TRUE & pscl == TRUE){
+    
+    rollcallObj_1 <- rollcall(rollCallMatrix)
+    Legis_data <- Leg
+    list_rc_pscl <- list(rollCallObject = rollcallObj_1, 
+                         Legis_data = Legis_data)
+    message("Returning list of rollcall object and legislator\ncharacteristics for pscl")
+    return(list_rc_pscl)
+  } else if(Legis == TRUE & Stan == TRUE){
+    
+    miss <- which(is.na(rollCallMatrix))
+    StanObject <- rollCallMatrix[-miss]
+    Legis_data <- Leg
+    list_rc_stan <- list(StanObject = StanObject, 
+                         Legis_data = Legis_data)
+    message("Returning list of rollcall vector and legislator \ncharacteristics for Stan")
+    return(list_rc_stan)
+    
+  } else if(pscl == FALSE & Stan == FALSE & Legis == FALSE){
```

Comment:
I'm the biggest @Alison-Li stan after this PR

Reply:
LOL I didn't even notice